api/hypershift/v1beta1/azure.go

```diff
@@ -271,10 +277,10 @@ const (
 	EphemeralDiskPersistence AzureDiskPersistence = "Ephemeral"
 )
 
-// +kubebuilder:validation:XValidation:rule="!has(self.diskStorageAccountType) || self.diskStorageAccountType != 'UltraSSD' || self.sizeGiB <= 32767",message="When not using storageAccountType UltraSSD, the SizeGB value must be less than or equal to 32,767"
+// +kubebuilder:validation:XValidation:rule="!has(self.diskStorageAccountType) || self.diskStorageAccountType != 'UltraSSD_LRS' || self.sizeGiB <= 32767",message="When not using diskStorageAccountType UltraSSD_LRS, the SizeGB value must be less than or equal to 32,767"
 type AzureNodePoolOSDisk struct {
 	// SizeGiB is the size in GiB (1024^3 bytes) to assign to the OS disk.
-	// This should be between 16 and 65,536 when using the UltraSSD storage account type and between 16 and 32,767 when using any other storage account type.
+	// This should be between 16 and 65,536 when using the UltraSSD_LRS storage account type and between 16 and 32,767 when using any other storage account type.
```

Comment:
According to https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison, it looks like the size could be up to 65536 when using both Ultra SSD and Premium SSD V2.